src/cache/file_cache.cc

```diff
@@ -179,12 +218,27 @@ bool FileCache::Find(UnhandledSource code, CommandLine command_line,
 void FileCache::Store(UnhandledSource code, CommandLine command_line,
                       Version version, const List<String>& headers,
                       const String& current_dir, string::HandledHash hash) {
-  DoStore(Hash(code, command_line, version), headers, current_dir, hash);
+  Store(code, List<Immutable>{}, command_line, version, headers, current_dir,
+        hash);
+}
+
+void FileCache::Store(UnhandledSource code, const List<Immutable>& extra_files,
```

Comment:
It's better to leave a single `Store` method: either put the `extra_files` in the end so they can be set to empty default, or simply pass empty brace-initializer everywhere throughout the code.